Framework/API/inc/MantidAPI/ComponentInfo.h

```diff
@@ -2,6 +2,8 @@
 #define MANTID_API_COMPONENTINFO_H_
 
 #include "MantidAPI/DllConfig.h"
+#include "MantidKernel/V3D.h"
+#include "MantidKernel/Quat.h"
```

Comment:
position() and rotation() return by value (as they do in `DetectorInfo`) so unfortunately we cannot forward declare.